activerecord/lib/active_record/associations/join_dependency.rb

```diff
@@ -227,8 +228,17 @@ def walk(left, right)
       end
 
       def find_reflection(klass, name)
-        klass._reflect_on_association(name) or
-          raise ConfigurationError, "Association named '#{ name }' was not found on #{ klass.name }; perhaps you misspelled it?"
```

Comment:
Would it be sufficient, to address the original complaint, to just use wording more like "Can't join to association named ..", without being specific about exactly which relation method was used? That seems to be an improvement in clarity, without needing to pass around extra data (or duplicate logic to parse it) just to construct an error message.